torchvision/csrc/io/decoder/subtitle_stream.cpp

```diff
@@ -43,21 +43,35 @@ int SubtitleStream::initFormat() {
 int SubtitleStream::analyzePacket(const AVPacket* packet, bool* gotFrame) {
   // clean-up
   releaseSubtitle();
-  // check flush packet
-  AVPacket avPacket;
-  av_init_packet(&avPacket);
-  avPacket.data = nullptr;
-  avPacket.size = 0;
-  auto pkt = packet ? *packet : avPacket;
+
+  auto pkt = packet;
+  if (pkt == nullptr) {
+    // check flush packet
+    AVPacket* avPacket;
+    avPacket = av_packet_alloc();
+    if (avPacket == nullptr) {
+      LOG(ERROR)
+          << "decoder as not able to allocate the subtitle-specific packet.";
+      return ENOMEM;
```

Comment:
Are we sure that this error code will work as return type for the method? Should we be returning one of the error codes of avcodec instead?

Reply:
I've updated them to the most appropriate [AVERROR](https://ffmpeg.org/doxygen/trunk/group__lavu__error.html)


Having done that, I'd say ENOMEM is more appropriate than AV alternative that I see used (`AVERROR_BUFFER_TOO_SMALL`).